lib/c/string.zig

```diff
@@ -5,7 +5,32 @@ const common = @import("common.zig");
 comptime {
     @export(&strcmp, .{ .name = "strcmp", .linkage = common.linkage, .visibility = common.visibility });
     @export(&strlen, .{ .name = "strlen", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strnlen, .{ .name = "strnlen", .linkage = common.linkage, .visibility = common.visibility });
     @export(&strncmp, .{ .name = "strncmp", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strchr, .{ .name = "strchr", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strchr, .{ .name = "index", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strrchr, .{ .name = "strrchr", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strrchr, .{ .name = "rindex", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strcpy, .{ .name = "strcpy", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strcat, .{ .name = "strcat", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strncpy, .{ .name = "strncpy", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&memccpy, .{ .name = "memccpy", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&mempcpy, .{ .name = "mempcpy", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&memmem, .{ .name = "memmem", .linkage = common.linkage, .visibility = common.visibility });
+    { // TODO: Conditional export for armeabi?
+        @export(&__aeabi_memclr, .{ .name = "__aeabi_memclr8", .linkage = common.linkage, .visibility = common.visibility });
+        @export(&__aeabi_memclr, .{ .name = "__aeabi_memclr4", .linkage = common.linkage, .visibility = common.visibility });
+        @export(&__aeabi_memclr, .{ .name = "__aeabi_memclr", .linkage = common.linkage, .visibility = common.visibility });
+    }
+    @export(&memchr, .{ .name = "memchr", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strchrnul, .{ .name = "__strchrnul", .linkage = .weak, .visibility = .hidden });
```

Comment:
These underscore-prefixed symbols look to be specific to musl and wasi-libc. They should be moved into a block guarded by `if (builtin.target.isMuslLibC() or builtin.target.isWasiLibC())`.